ZeroDivisionError: division by zero on API calls

Hi,

When running some tests with kytos_stats, I found this error:

```
        api_url = KYTOS_STATS + f'/bytes_count/{flow_id}'
        response = requests.get(api_url)
>       assert response.status_code == 200, response.text
E       AssertionError: Traceback (most recent call last):
E           File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/errors.py", line 162, in __call__
E             await self.app(scope, receive, _send)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/cors.py", line 84, in __call__
E             await self.app(scope, receive, send)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/exceptions.py", line 79, in __call__
E             raise exc
E           File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/exceptions.py", line 68, in __call__
E             await self.app(scope, receive, sender)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 716, in __call__
E             await route.handle(scope, receive, send)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 276, in handle
E             await self.app(scope, receive, send)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 68, in app
E             response = await run_in_threadpool(func, request)
E           File "/usr/local/lib/python3.9/dist-packages/starlette/concurrency.py", line 41, in run_in_threadpool
E             return await anyio.to_thread.run_sync(func, *args)
E           File "/usr/local/lib/python3.9/dist-packages/anyio/to_thread.py", line 31, in run_sync
E             return await get_asynclib().run_sync_in_worker_thread(
E           File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
E             return await future
E           File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 867, in run
E             result = context.run(func, *args)
E           File "//var/lib/kytos/napps/amlight/kytos_stats/main.py", line 130, in bytes_count
E             flow.stats.byte_count * 8 / flow.stats.duration_sec
E         ZeroDivisionError: division by zero
E
E       assert 500 == 200
E        +  where 500 = <Response [500]>.status_code
```


the test consists basically of installing a flow, waiting around 20 seconds and calling the API `/amlight/kytos_stats/v1/bytes_count/{flow_id}` (of_core.STATS_INTERVAL configured to 7 seconds)

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -22,6 +22,7 @@ Removed
 
 Fixed
 =====
+- Fixed error due to division by zero when ``duration_sec` is zero in the ``stats`` of a flow.
 
 Security
 ========

--- a/main.py
+++ b/main.py
@@ -108,11 +108,15 @@ class Main(KytosNApp):
         flow = self.flow_from_id(flow_id)
         if flow is None:
             raise HTTPException(404, detail="Flow does not exist")
+        try:
+            packet_per_second = \
+                flow.stats.packet_count / flow.stats.duration_sec
+        except ZeroDivisionError:
+            packet_per_second = 0
         packet_stats = {
             'flow_id': flow_id,
             'packet_counter': flow.stats.packet_count,
-            'packet_per_second':
-                flow.stats.packet_count / flow.stats.duration_sec
+            'packet_per_second': packet_per_second
             }
         return JSONResponse(packet_stats)
 
@@ -123,11 +127,15 @@ class Main(KytosNApp):
         flow = self.flow_from_id(flow_id)
         if flow is None:
             raise HTTPException(404, detail="Flow does not exist")
+        try:
+            bits_per_second = \
+                flow.stats.byte_count * 8 / flow.stats.duration_sec
+        except ZeroDivisionError:
+            bits_per_second = 0
         bytes_stats = {
             'flow_id': flow_id,
             'bytes_counter': flow.stats.byte_count,
-            'bits_per_second':
-                flow.stats.byte_count * 8 / flow.stats.duration_sec
+            'bits_per_second': bits_per_second
             }
         return JSONResponse(bytes_stats)
 
@@ -176,7 +184,10 @@ class Main(KytosNApp):
             if total:
                 count_flows += count
             else:
-                per_second = count / stats['duration_sec']
+                try:
+                    per_second = count / stats['duration_sec']
+                except ZeroDivisionError:
+                    per_second = 0
                 if rate.startswith('bits'):
                     per_second *= 8
                 count_flows.append({'flow_id': flow_id,

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -474,3 +474,30 @@ class TestMain:
         self.napp.handle_table_stats_reply_received(tables_mock)
         table = list(self.napp.tables_stats_dict.values())[0]
         assert list(table.keys())[0] == 10
+
+    @patch("napps.amlight.kytos_stats.main.Main.flow_stats_by_dpid_flow_id")
+    async def test_flows_counters_div_zero(self, mock_from_flow):
+        """Test that there is no error due to division by zero."""
+        flow_info = {
+            "byte_count": 10,
+            "packet_count": 20,
+            "duration_sec": 0
+            }
+        flow_id = '6055f13593fad45e0b4699f49d56b105'
+        flow_stats_dict_mock = {flow_id: flow_info}
+        dpid = "00:00:00:00:00:00:00:01"
+        flow_by_sw = {dpid: flow_stats_dict_mock}
+        mock_from_flow.return_value = flow_by_sw
+
+        self._patch_switch_flow(flow_id)
+        endpoint = f"{self.base_endpoint}/packet_count/per_flow/{dpid}"
+        response = await self.api_client.get(endpoint)
+        response = response.json()
+        assert response[0]["flow_id"] == flow_id
+        assert response[0]["packet_per_second"] == 0
+
+        endpoint = f"{self.base_endpoint}/bytes_count/per_flow/{dpid}"
+        response = await self.api_client.get(endpoint)
+        response = response.json()
+        assert response[0]["flow_id"] == flow_id
+        assert response[0]["bits_per_second"] == 0

